File: DSA/src/go/binary_tree_right_side_view.go
package main

func rightSideView(root *TreeNode) []int {
	if root == nil {
		return []int{}
	}
	queue := make(chan *TreeNode, 100)
	queue <- root
	var side []int
	var size, i int
	var node *TreeNode
	for len(queue) > 0 {
		size = len(queue)
		for i = 0; i < size; i++ {
			node = <-queue
			if i == size-1 {
				side = append(side, node.Val)
			}
			if node.Left != nil {
				queue <- node.Left
			}
			if node.Right != nil {
				queue <- node.Right
			}
		}
	}
	return side
}

File: docs/DSA/src/py/binary_tree_right_side_view.py
from collections import deque
from typing import Optional, List


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        if not root: return []
        q = deque([root])
        side = []
        while len(q) > 0:
            size = len(q)
            for i in range(size):
                node = q.pop()
                if i == size-1:
                    side.append(node.val)
                if node.left:
                    q.appendleft(node.left)
                if node.right:
                    q.appendleft(node.right)
        return side